tensorflow_addons/image/transform_ops_test.py

```diff
@@ -72,59 +72,38 @@ def test_transform_static_output_shape(self):
             output_shape=tf.constant([3, 5]))
         self.assertAllEqual([3, 5], result.shape)
 
-    def _test_grad(self, shape_to_test):
-        with self.cached_session():
-            test_image_shape = shape_to_test
-            test_image = np.random.randn(*test_image_shape)
-            test_image_tensor = tf.constant(test_image, shape=test_image_shape)
-            test_transform = transform_ops.angles_to_projective_transforms(
-                np.pi / 2, 4, 4)
+    def _test_grad(self, input_shape, output_shape=None):
+        test_image = tf.random.normal(input_shape, dtype=tf.float32)
 
-            output_shape = test_image_shape
-            output = transform_ops.transform(test_image_tensor, test_transform)
-            left_err = tf.compat.v1.test.compute_gradient_error(
-                test_image_tensor,
-                test_image_shape,
-                output,
-                output_shape,
-                x_init_value=test_image)
-            self.assertLess(left_err, 1e-10)
+        if output_shape is None:
+            resize_shape = None
+        elif len(output_shape) == 2:
+            resize_shape = output_shape
+        elif len(output_shape) == 3:
+            resize_shape = output_shape[0:2]
+        elif len(output_shape) == 4:
+            resize_shape = output_shape[1:3]
 
-    def _test_grad_different_shape(self, input_shape, output_shape):
-        with self.cached_session():
-            test_image_shape = input_shape
-            test_image = np.random.randn(*test_image_shape)
-            test_image_tensor = tf.constant(test_image, shape=test_image_shape)
-            test_transform = transform_ops.angles_to_projective_transforms(
+        def transform_fn(x):
+            x.set_shape(input_shape)
+            transform = transform_ops.angles_to_projective_transforms(
                 np.pi / 2, 4, 4)
+            return transform_ops.transform(
+                images=x, transforms=transform, output_shape=resize_shape)
+
+        theoretical, numerical = tf.test.compute_gradient(
```

Comment:
Tzu-Wei, take the private op `max_error` a try, and see if it works?

https://github.com/tensorflow/tensorflow/blob/0ff7955a0c1a42e2767afb0a5cc202dfe4d6ff19/tensorflow/python/kernel_tests/relu_op_test.py#L119

Reply:
The reported error from `max_error` is 7.25388e-5... Still exceeds original tolerance 1e-10.